src/sentry/sentry_apps/services/app/impl.py

```diff
@@ -61,20 +61,24 @@ def get_many(
     def find_app_components(self, *, app_id: int) -> list[RpcSentryAppComponent]:
         return [
             serialize_sentry_app_component(c)
-            for c in SentryAppComponent.objects.filter(sentry_app_id=app_id)
+            for c in SentryAppComponent.objects.get_from_cache()
+            .using_replica()
+            .filter(sentry_app_id=app_id)
         ]
 
     def get_sentry_app_by_id(self, *, id: int) -> RpcSentryApp | None:
         try:
-            sentry_app = SentryApp.objects.get(id=id)
+            sentry_app = SentryApp.objects.get_from_cache().using_replica().get(id=id)
```

Comment:
```suggestion
            sentry_app = SentryApp.objects.get_from_cache(id=id, use_replica=True)
```

I think you want this, so you can read from cache or fallback to a replica read.